kmip/services/kmip_client.py

```diff
@@ -217,6 +217,7 @@ def open(self):
             suppress_ragged_eofs=self.suppress_ragged_eofs)
         self.protocol = KMIPProtocol(self.socket)
 
+        self.socket.settimeout(self.timeout)
         self.socket.connect((self.host, self.port))
```

Comment:
Do we want to catch a socket exception here, log and raise it. This helps two things i believe, one for audit in the log for connection issues, and one predictable exception code consumers can expect and abstracted from underlying socket codes.